stl/inc/algorithm

```diff
@@ -1865,6 +1865,199 @@ _NODISCARD _FwdIt1 find_end(_ExPo&& _Exec, _FwdIt1 _First1, _FwdIt1 _Last1, _Fwd
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // CLASS ranges::_Find_end_fn
+    class _Find_end_fn : private _Not_quite_object {
+    private:
+        template <class _It1, class _It2, class _Pr, class _Pj1, class _Pj2>
+        _NODISCARD static constexpr bool _Equal_rev_pred_unchecked(
+            _It1 _First1, _It2 _First2, const _It2 _Last2, _Pr _Pred, _Pj1 _Proj1, _Pj2 _Proj2) {
+            // compare [_First1, ...) to [_First2, _Last2)
+            constexpr bool _Both_identity = is_same_v<_Pj1, identity> && is_same_v<_Pj2, identity>;
+            // Performance note: teach _Equal_memcmp_is_safe about contiguous iterators and ranges::equal_to
+            if constexpr (_Both_identity && decltype(_Equal_memcmp_is_safe(_First1, _First2, _Pred))::value) {
+#ifdef __cpp_lib_is_constant_evaluated

```

Comment:
why do we still need this check? is it for clang-cl?

Reply:
Good question - I don't remember. For all I know this isn't necessary in `_HAS_CXX20`.

Reply:
Yes: we define this unconditionally in C++20 mode now; cleaning.

Reply:
Found another occurrence in `lexicographical_compare_three_way` in `<xutility>` that I could remove as well.